folding-schemes/src/utils/gadgets.rs

```diff
@@ -15,6 +15,11 @@ pub fn mat_vec_mul_sparse<F: PrimeField, CF: PrimeField, FV: FieldVar<F, CF>>(
     let mut res = vec![FV::zero(); m.n_rows];
     for (row_i, row) in m.coeffs.iter().enumerate() {
         for (value, col_i) in row.iter() {
+            if value.value().unwrap() == F::one() {
```

Comment:
Since the circuit that does this is using the A,B,C matrices as constants, this is a logic 'hardcoded' in the circuit, so it is 'translated' into static logic in the circuit and it should not be a problem, right?
How I see it is, imagine that we write the specific constraints one by one by hand to verify a set of specific R1CS matrices, and if we want to use another set of R1CS matrices we need to rewrite the specific constraints that verify that. In this case we avoid writing all the specific constraints one by one thanks to being able to do the loop.